internal/dataplane/kong_client_konnect.go

```diff
@@ -0,0 +1,103 @@
+package dataplane
+
+import (
+	"context"
+	"errors"
+
+	"github.com/go-logr/logr"
+	"github.com/kong/go-kong/kong"
+
+	"github.com/kong/kubernetes-ingress-controller/v3/internal/adminapi"
+	"github.com/kong/kubernetes-ingress-controller/v3/internal/clients"
+	"github.com/kong/kubernetes-ingress-controller/v3/internal/dataplane/deckerrors"
+	"github.com/kong/kubernetes-ingress-controller/v3/internal/dataplane/kongstate"
+	"github.com/kong/kubernetes-ingress-controller/v3/internal/dataplane/sendconfig"
+)
+
+// maybeSendOutToKonnectClient sends out the configuration to Konnect when KonnectClient is provided.
+// It's a noop when Konnect integration is not enabled.
+func (c *KongClient) maybeSendOutToKonnectClient(
+	ctx context.Context,
+	s *kongstate.KongState,
+	config sendconfig.Config,
+	isFallback bool,
+) {
+	konnectClient := c.clientsProvider.KonnectClient()
+	// There's no KonnectClient configured, that's totally fine.
+	if konnectClient == nil {
+		return
+	}
+
+	// send Kong configuration to Konnect client in a new goroutine
+	go c.sendOutToKonnectClient(
+		ctx, konnectClient, s, config, isFallback,
+	)
+}
```

Comment:
I think move the goroutine spawn out to the call within `Update()` with a comment mentioning the goroutine's lifecycle (AFAIK it can't hang around, so just note that it'll either terminate after successfully updating or failing and marking status appropriately). Makes it slightly more readable within `Update()` for someone scanning over it to understand that the Konnect update is being forked off in its own routine.